src/transformers/modeling_utils.py

```diff
@@ -2215,30 +2231,25 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
                 if low_cpu_mem_usage is None:
                     low_cpu_mem_usage = True
 
-        if load_in_8bit:
+        if load_in_8bit or load_in_4bit:
             if not (is_accelerate_available() and is_bitsandbytes_available()):
                 raise ImportError(
                     "Using `load_in_8bit=True` requires Accelerate: `pip install accelerate` and the latest version of"
                     " bitsandbytes `pip install -i https://test.pypi.org/simple/ bitsandbytes` or"
                     " pip install bitsandbytes` "
                 )
-            if torch_dtype != torch.float16:
-                # We force the `dtype` to be float16, this is a requirement from `bitsandbytes`
-                logger.warning(
-                    f"Overriding torch_dtype={torch_dtype} with `torch_dtype=torch.float16` due to "
-                    "requirements of `bitsandbytes` to enable model loading in mixed int8. "
-                    "Either pass torch_dtype=torch.float16 or don't pass this argument at all to remove this warning."
-                )
-                torch_dtype = torch.float16
```

Comment:
As in the Accelerate PR, is this not necessary anymore in the current released version?

Reply:
It was never needed in the first place. The problem with this is that it does not allow for other data types which are necessary for mixed-precision 4-bit training.

Reply:
Added a proper warning message!